libraries/botbuilder-core/botbuilder/core/adapters/test_adapter.py

```diff
@@ -151,10 +177,71 @@ def id_mapper(activity):
                 if len(arg) == 4:
                     timeout = arg[3]
             await self.test(arg[0], arg[1], description, timeout)
+    
+    def add_user_token(self, connection_name: str, channel_id: str, user_id: str, token: str, magic_code: str = None):
+        key = UserToken()
+        key.channel_id = channel_id
+        key.connection_name = connection_name
+        key.user_id = user_id
+        key.token = token
+
+        if not magic_code:
+            self._user_tokens.append(key)
+        else:
+            mc = TokenMagicCode()
+            mc.key = key
+            mc.magic_code = magic_code
+            self._magic_codes.append(mc)
+
+    async def get_user_token(self, context: TurnContext, connection_name: str, magic_code: str= None) -> TokenResponse:
```

Comment:
missing space on last parameter

Reply:
I see many different patterns in this file:

str= None
str = None
Coroutine=None

But the pattern should be the exact same in all files :)

Reply:
solved